homeassistant/components/zwave_js/__init__.py

```diff
@@ -131,13 +131,15 @@ def async_on_node_removed(node: ZwaveNode) -> None:
         # grab device in device registry attached to this node
         dev_id = get_device_id(client, node)
         device = dev_reg.async_get_device({dev_id})
-        # note: removal of entity registry is handled by core
+        assert device
```

Comment:
MyPy complains because `async_get_device` is not guaranteed to return a device registry entry.
Is some actual error handling needed to validate the return value?

Reply:
```suggestion
        if not device:
            return
```

Reply:
Modified to explicitly silence the MyPy errors for now, let's do proper error handling + tests in a separate PR